src/Hooks/UseServiceQuery.ts

```diff
@@ -118,32 +107,29 @@ export const useServiceQuery = <
         context,
         operationName,
       })
-      const exception = new ServiceErrorException({
+      const exception = new ServiceOperationError({
         serviceErrorList: errorList,
-        serviceName: operationName,
+        operationName,
         context,
       })
       return exception
     }
     return null
-  }, [context, memoizedQuery, queryDocument])
+  }, [context, memoizedQuery.error, queryDocument])
   useLayoutEffect(() => {
-    if ((exception != null) && (shownExceptionListRef.current.find(shownException => (
-      isServiceErrorListEqual(shownException.serviceErrorList, exception.serviceErrorList)
-    )) == null)) {
-      addServiceErrorException(exception)
-      shownExceptionListRef.current.push(exception)
-    }
-    return () => {
-      (exception != null) && removeServiceErrorException(context)
+    if ((exception != null) && (reportedOperationErrorListRef.current.includes(exception))) {
+      reportError(exception)
+      reportedOperationErrorListRef.current.push(exception)
     }
-  }, [addServiceErrorException, context, exception, memoizedVariables, queryDocument, removeServiceErrorException])
+  }, [context, exception, memoizedVariables, queryDocument])
 
   const promiselessRefetch = useCallback((...args: Parameters<typeof memoizedQuery.refetch>) => {
+    reportedOperationErrorListRef.current = []
     asyncToCallback(memoizedQuery.refetch(...args))
```

Comment:
we need to check if failed refetch includes the error in query result

Reply:
Error is added to `query.exception` when using refetch.